### PR TITLE
DDST-373: Add new fields to FOXML migration

### PR DESCRIPTION
**Pull request by chrismacdonaldw**

### CODE REVIEW

File: modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml [hunk 1]
```diff
@@ -1693,6 +1693,36 @@ process:
           - plugin: callback
             callable: array_filter
           - plugin: null_coalesce
+        field_related_item_identifier:
+          - <<: *nested_mods_node
+            query: 'mods:relatedItem/mods:identifier'
```
[the remark this answers is not in the document]
Comment (adam-vessey) [reply modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml:1698]: Have this in https://github.com/discoverygarden/dgi_migrate/pull/135

File: modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml [hunk 2]
```diff
@@ -1693,6 +1693,36 @@ process:
...
+            property: nodeValue
+          - plugin: callback
+            callable: trim
+          - plugin: single_value
+          - plugin: callback
+            callable: array_filter
+          - plugin: null_coalesce
+        field_related_item_genre:
+          - <<: *nested_mods_node
+            query: 'mods:relatedItem/mods:genre'
```
[the remark this answers is not in the document]
Comment (adam-vessey) [reply modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml:1712]: Have this in https://github.com/discoverygarden/dgi_migrate/pull/135

File: modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml [hunk 3]
```diff
@@ -1693,6 +1693,36 @@ process:
...
+            method: process
+          - plugin: multiple_values
+          - plugin: dgi_migrate.sub_process
+            process_values: true
+            values:
+              _vid:
+                - plugin: default_value
+                  default_value: genre
+              <<: *generic_term_after
+          - <<: *generic_term_extract
```
[the remark this answers is not in the document]
Comment (adam-vessey) [reply modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml:1725]: Addressed in https://github.com/discoverygarden/dgi_migrate/pull/135

File: modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml [hunk 4]
```diff
@@ -1973,6 +2003,455 @@ process:
...
+    - plugin: multiple_values
+    - plugin: dgi_migrate.subproperty
+      property: nodeValue
+    - plugin: single_value
+    - plugin: callback
+      callable: array_filter
+    - plugin: null_coalesce
+  field_funder:
+    - <<: *base_mods_node
+      query: 'mods:role/mods:roleTerm="funder"'
```
[the remark this answers is not in the document]
Comment (adam-vessey) [reply modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml:2408]: Adjusted XPath in https://github.com/discoverygarden/dgi_migrate/pull/135 , but yeah, not sure about more general corporate name things. In general, duplicating data seems like a bad idea, but whatever.

File: modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml [hunk 5]
```diff
@@ -1973,6 +2003,455 @@ process:
...
+    - plugin: multiple_values
+    - plugin: dgi_migrate.subproperty
+      property: nodeValue
+    - plugin: single_value
+    - plugin: callback
+      callable: array_filter
+    - plugin: null_coalesce
+  _use_license:
+    - <<: *base_mods_node
+      query: 'mods:accessCondition[@type="use and reproduction"][@displayLabe="Creative Commons"]'
```
Comment (adam-vessey) [inline modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml:2436]: I know it's such in the spec, but, seems like a typo?:

```suggestion
      query: 'mods:accessCondition[@type="use and reproduction"][@displayLabel="Creative Commons"]'
```

**Review by adam-vessey — COMMENTED** [review]
Got a number of fixes together in https://github.com/discoverygarden/dgi_migrate/pull/135 but it seems like there's other issues that are unaccounted for, such as `field_subject` now allowing only `subject` and `geo_location` term references, but the current mapping of `subject`, `person` and `organization` doesn't seem to be flagged as a gap in the MMS.

File: modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (adam-vessey) [reply modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml]: Discussion w/ @chrisDay50: Things should go to the "corporate body" taxonomy over the "funders" taxonomy. Copypasta'd to make it so.

File: modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml (hunk 5 above)
Comment (adam-vessey) [reply modules/dgi_migrate_foxml_standard_mods/migrations/dgis_nodes.yml]: Made it so in #135

**Review by nchiasson-dgi — APPROVED** [review]

**Pull request merged**